src/nfa.ts

```diff
@@ -117,6 +116,73 @@ export class NFA implements ReadonlyNFA {
 		return match(0, nodes.initial);
 	}
 
+	// Thompson simulation NFA matching from Cox article
+	match(input: string) {
```

Comment:
The input has to be an `Iterable<number>` (or a `readonly number[]` if you need random access or iterate over it multiple times).

By taking a list of numbers instead of a string, we don't have to convert the string into code points. This is important because we don't know __how__ the string should be converted (UTF16 characters, Unicode code points, something else).